django/core/signing.py

```diff
@@ -150,11 +155,23 @@ class Signer(object):
 
     def __init__(self, key=None, sep=':', salt=None):
         # Use of native strings in all versions of Python
-        self.sep = force_str(sep)
+        self.sep = sep
         self.key = key or settings.SECRET_KEY
         self.salt = force_str(salt or
             '%s.%s' % (self.__class__.__module__, self.__class__.__name__))
 
+    @property
+    def sep(self):
+        return self._sep
+
+    @sep.setter
+    def sep(self, value):
+        sep = force_str(value)
+        if _SEP_UNSAFE.match(sep):
+            warnings.warn('Unsafe Signer separator: %r (cannot be empty or consist of only A-z0-9-_=)' % sep,
```

Comment:
are you sure about this point of "consist of only A-z0-9-_=", as opposed to those characters may not appear in the separator? (not too familiar with this restriction myself)
I don't know that the complexity of a `setter` is necessary. I'd just modify the test to initialize a `Signer` each time.